File: Syncfusion_Sample/Windows/Tools.Windows/Samples/2.0/Product Showcase/VS Demo/VB/NewFileDialog.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace VSDemo
	''' <summary>
	''' Summary description for NewFileDialog.
	''' </summary>
	Public Class NewFileDialog : Inherits System.Windows.Forms.Form

		Private mainForm As VSDemoMainForm

		Private WithEvents listView1 As System.Windows.Forms.ListView
		Private label1 As System.Windows.Forms.Label
		Private label2 As System.Windows.Forms.Label
		Private textBox1 As System.Windows.Forms.TextBox
        Private Shadows WithEvents okButton As System.Windows.Forms.Button
        Private Shadows WithEvents cancelButton As System.Windows.Forms.Button
		Private dockingImageList As System.Windows.Forms.ImageList
		Private components As System.ComponentModel.IContainer

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal mainForm As VSDemoMainForm)
			InitializeComponent()
			Me.mainForm = mainForm
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.components = New System.ComponentModel.Container()
			Dim listViewItem1 As System.Windows.Forms.ListViewItem = New System.Windows.Forms.ListViewItem(New String() { ".cs"}, 63, System.Drawing.SystemColors.WindowText, System.Drawing.SystemColors.Window, New System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, (CByte(0))))
			Dim listViewItem2 As System.Windows.Forms.ListViewItem = New System.Windows.Forms.ListViewItem(New String() { ".vb"}, 4, System.Drawing.SystemColors.WindowText, System.Drawing.SystemColors.Window, New System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, (CByte(0))))
			Dim resources As System.ComponentModel.ComponentResourceManager = New System.ComponentModel.ComponentResourceManager(GetType(NewFileDialog))
			Me.listView1 = New System.Windows.Forms.ListView()
			Me.dockingImageList = New System.Windows.Forms.ImageList(Me.components)
			Me.label1 = New System.Windows.Forms.Label()
			Me.label2 = New System.Windows.Forms.Label()
			Me.textBox1 = New System.Windows.Forms.TextBox()
			Me.okButton = New System.Windows.Forms.Button()
			Me.cancelButton = New System.Windows.Forms.Button()
			Me.SuspendLayout()
			' 
			' listView1
			' 
			Me.listView1.Items.AddRange(New System.Windows.Forms.ListViewItem() { listViewItem1, listViewItem2})
			Me.listView1.LargeImageList = Me.dockingImageList
			Me.listView1.Location = New System.Drawing.Point(8, 48)
			Me.listView1.Name = "listView1"
			Me.listView1.Size = New System.Drawing.Size(480, 160)
			Me.listView1.SmallImageList = Me.dockingImageList
			Me.listView1.TabIndex = 0
			Me.listView1.UseCompatibleStateImageBehavior = False
'			Me.listView1.SelectedIndexChanged += New System.EventHandler(Me.listView1_SelectedIndexChanged);
			' 
			' dockingImageList
			' 
			Me.dockingImageList.ImageStream = (CType(resources.GetObject("dockingImageList.ImageStream"), System.Windows.Forms.ImageListStreamer))
			Me.dockingImageList.TransparentColor = System.Drawing.Color.Transparent
			Me.dockingImageList.Images.SetKeyName(0, "")
			Me.dockingImageList.Images.SetKeyName(1, "")
			Me.dockingImageList.Images.SetKeyName(2, "")
			Me.dockingImageList.Images.SetKeyName(3, "")
			Me.dockingImageList.Images.SetKeyName(4, "")
			Me.dockingImageList.Images.SetKeyName(5, "")
			Me.dockingImageList.Images.SetKeyName(6, "")
			Me.dockingImageList.Images.SetKeyName(7, "")
			Me.dockingImageList.Images.SetKeyName(8, "")
			Me.dockingImageList.Images.SetKeyName(9, "")
			Me.dockingImageList.Images.SetKeyName(10, "")
			Me.dockingImageList.Images.SetKeyName(11, "")
			Me.dockingImageList.Images.SetKeyName(12, "")
			Me.dockingImageList.Images.SetKeyName(13, "")
			Me.dockingImageList.Images.SetKeyName(14, "")
			Me.dockingImageList.Images.SetKeyName(15, "")
			Me.dockingImageList.Images.SetKeyName(16, "")
			Me.dockingImageList.Images.SetKeyName(17, "")
			Me.dockingImageList.Images.SetKeyName(18, "")
			Me.dockingImageList.Images.SetKeyName(19, "")
			Me.dockingImageList.Images.SetKeyName(20, "")
			Me.dockingImageList.Images.SetKeyName(21, "")
			Me.dockingImageList.Images.SetKeyName(22, "")
			Me.dockingImageList.Images.SetKeyName(23, "")
			Me.dockingImageList.Images.SetKeyName(24, "")
			Me.dockingImageList.Images.SetKeyName(25, "")
			Me.dockingImageList.Images.SetKeyName(26, "")
			Me.dockingImageList.Images.SetKeyName(27, "")
			Me.dockingImageList.Images.SetKeyName(28, "")
			Me.dockingImageList.Images.SetKeyName(29, "")
			Me.dockingImageList.Images.SetKeyName(30, "")
			Me.dockingImageList.Images.SetKeyName(31, "")
			Me.dockingImageList.Images.SetKeyName(32, "")
			Me.dockingImageList.Images.SetKeyName(33, "")
			Me.dockingImageList.Images.SetKeyName(34, "")
			Me.dockingImageList.Images.SetKeyName(35, "")
			Me.dockingImageList.Images.SetKeyName(36, "")
			Me.dockingImageList.Images.SetKeyName(37, "")
			Me.dockingImageList.Images.SetKeyName(38, "")
			Me.dockingImageList.Images.SetKeyName(39, "")
			Me.dockingImageList.Images.SetKeyName(40, "")
			Me.dockingImageList.Images.SetKeyName(41, "")
			Me.dockingImageList.Images.SetKeyName(42, "")
			Me.dockingImageList.Images.SetKeyName(43, "")
			Me.dockingImageList.Images.SetKeyName(44, "")
			Me.dockingImageList.Images.SetKeyName(45, "")
			Me.dockingImageList.Images.SetKeyName(46, "")
			Me.dockingImageList.Images.SetKeyName(47, "")
			Me.dockingImageList.Images.SetKeyName(48, "")
			Me.dockingImageList.Images.SetKeyName(49, "")
			Me.dockingImageList.Images.SetKeyName(50, "")
			Me.dockingImageList.Images.SetKeyName(51, "")
			Me.dockingImageList.Images.SetKeyName(52, "")
			Me.dockingImageList.Images.SetKeyName(53, "")
			Me.dockingImageList.Images.SetKeyName(54, "")
			Me.dockingImageList.Images.SetKeyName(55, "")
			Me.dockingImageList.Images.SetKeyName(56, "")
			Me.dockingImageList.Images.SetKeyName(57, "")
			Me.dockingImageList.Images.SetKeyName(58, "")
			Me.dockingImageList.Images.SetKeyName(59, "")
			Me.dockingImageList.Images.SetKeyName(60, "")
			Me.dockingImageList.Images.SetKeyName(61, "")
			Me.dockingImageList.Images.SetKeyName(62, "")
			Me.dockingImageList.Images.SetKeyName(63, "")
			Me.dockingImageList.Images.SetKeyName(64, "")
			Me.dockingImageList.Images.SetKeyName(65, "")
			Me.dockingImageList.Images.SetKeyName(66, "")
			Me.dockingImageList.Images.SetKeyName(67, "")
			Me.dockingImageList.Images.SetKeyName(68, "")
			Me.dockingImageList.Images.SetKeyName(69, "")
			Me.dockingImageList.Images.SetKeyName(70, "")
			Me.dockingImageList.Images.SetKeyName(71, "")
			Me.dockingImageList.Images.SetKeyName(72, "")
			Me.dockingImageList.Images.SetKeyName(73, "")
			Me.dockingImageList.Images.SetKeyName(74, "")
			Me.dockingImageList.Images.SetKeyName(75, "")
			Me.dockingImageList.Images.SetKeyName(76, "")
			Me.dockingImageList.Images.SetKeyName(77, "")
			Me.dockingImageList.Images.SetKeyName(78, "")
			Me.dockingImageList.Images.SetKeyName(79, "")
			Me.dockingImageList.Images.SetKeyName(80, "")
			' 
			' label1
			' 
			Me.label1.Location = New System.Drawing.Point(8, 16)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(100, 23)
			Me.label1.TabIndex = 1
			Me.label1.Text = "Templates"
			' 
			' label2
			' 
			Me.label2.Location = New System.Drawing.Point(8, 224)
			Me.label2.Name = "label2"
			Me.label2.Size = New System.Drawing.Size(64, 23)
			Me.label2.TabIndex = 2
			Me.label2.Text = "FileName"
			' 
			' textBox1
			' 
			Me.textBox1.Location = New System.Drawing.Point(8, 248)
			Me.textBox1.Name = "textBox1"
			Me.textBox1.Size = New System.Drawing.Size(472, 23)
			Me.textBox1.TabIndex = 3
			' 
			' okButton
			' 
			Me.okButton.BackColor = System.Drawing.SystemColors.Control
			Me.okButton.Location = New System.Drawing.Point(312, 280)
			Me.okButton.Name = "okButton"
			Me.okButton.Size = New System.Drawing.Size(75, 23)
			Me.okButton.TabIndex = 4
			Me.okButton.Text = "Ok"
			Me.okButton.UseVisualStyleBackColor = False
'			Me.okButton.Click += New System.EventHandler(Me.okButton_Click);
			' 
			' cancelButton
			' 
			Me.cancelButton.Location = New System.Drawing.Point(408, 280)
			Me.cancelButton.Name = "cancelButton"
			Me.cancelButton.Size = New System.Drawing.Size(75, 23)
			Me.cancelButton.TabIndex = 5
			Me.cancelButton.Text = "Cancel"
'			Me.cancelButton.Click += New System.EventHandler(Me.cancelButton_Click);
			' 
			' NewFileDialog
			' 
			Me.AutoScaleBaseSize = New System.Drawing.Size(6, 16)
			Me.ClientSize = New System.Drawing.Size(496, 334)
			Me.Controls.Add(Me.cancelButton)
			Me.Controls.Add(Me.okButton)
			Me.Controls.Add(Me.textBox1)
			Me.Controls.Add(Me.label2)
			Me.Controls.Add(Me.label1)
			Me.Controls.Add(Me.listView1)
			Me.Font = New System.Drawing.Font("Calibri", 9.75F)
			Me.Name = "NewFileDialog"
			Me.Text = "NewFileDialog"
			Me.ResumeLayout(False)
			Me.PerformLayout()

		End Sub
		#End Region

        Private Sub listView1_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles listView1.SelectedIndexChanged
            If Me.listView1.SelectedItems.Count > 0 Then
                If Me.listView1.SelectedItems(0).Text <> "" Then
                    If Me.listView1.SelectedItems(0).Text = ".cs" Then
                        Me.textBox1.Text = "Class1.cs"
                        Me.textBox1.Focus()
                    ElseIf Me.listView1.SelectedItems(0).Text = ".vb" Then
                        Me.textBox1.Text = "Class1.vb"
                        Me.textBox1.Focus()
                    End If
                End If
            End If
        End Sub

		Private Sub okButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles okButton.Click
			If textBox1.Text<>"" Then
				If textBox1.Text.EndsWith(".cs") OrElse textBox1.Text.EndsWith(".vb") Then
					If (Not System.IO.File.Exists(System.IO.Path.GetDirectoryName(Application.ExecutablePath) & "\" & textBox1.Text)) Then
						createNewFile()
					Else
						Dim dr As DialogResult = MessageBox.Show("File already exists,Overwrite it??","",MessageBoxButtons.OKCancel,MessageBoxIcon.Information)
                        If dr = Windows.Forms.DialogResult.OK Then
                            createNewFile()
                        End If
					End If
				Else
					MessageBox.Show("Enter proper extension for File")
				End If
			End If
		End Sub

		Private Sub createNewFile()
			Try
				Dim activeMDIChild As Form = Me.mainForm.CheckFormInChildWindow(Me.textBox1.Text)
				If Not activeMDIChild Is Nothing Then
					activeMDIChild.Close()
				End If
				System.IO.File.Create(System.IO.Path.GetDirectoryName(Application.ExecutablePath) & "\" & textBox1.Text)
				mainForm.openFileInCodeEditor(System.IO.Path.GetDirectoryName(Application.ExecutablePath) & "\" & textBox1.Text)
				Me.Close()
			Catch e As Exception
				MessageBox.Show(e.Message)
			End Try
		End Sub

		Private Sub cancelButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles cancelButton.Click
			Me.Close()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs)
			Me.listView1.View = View.LargeIcon
		End Sub

		Private Sub button2_Click(ByVal sender As Object, ByVal e As System.EventArgs)
			Me.listView1.View = View.SmallIcon
		End Sub
	End Class
End Namespace

File: Syncfusion_Sample/Windows/Tools.Windows/Samples/2.0/Product Showcase/VS Demo/CS/NewFileDialog.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace VSDemo
{
    /// <summary>
    /// Summary description for NewFileDialog.
    /// </summary>
    public class NewFileDialog : System.Windows.Forms.Form
    {

        private VSDemoMainForm mainForm;

        private System.Windows.Forms.ListView listView1;
        private System.Windows.Forms.Label label1;
        private System.Windows.Forms.Label label2;
        private System.Windows.Forms.TextBox textBox1;
        private System.Windows.Forms.Button okButton;
        private System.Windows.Forms.Button cancelButton;
        private System.Windows.Forms.ImageList dockingImageList;
        private System.ComponentModel.IContainer components;

        public NewFileDialog()
        {
            InitializeComponent();
        }

        public NewFileDialog(VSDemoMainForm mainForm)
        {
            InitializeComponent();
            this.mainForm = mainForm;
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose(disposing);
        }

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container();
            System.Windows.Forms.ListViewItem listViewItem1 = new System.Windows.Forms.ListViewItem(new string[] {
            ".cs"}, 63, System.Drawing.SystemColors.WindowText, System.Drawing.SystemColors.Window, new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0))));
            System.Windows.Forms.ListViewItem listViewItem2 = new System.Windows.Forms.ListViewItem(new string[] {
            ".vb"}, 4, System.Drawing.SystemColors.WindowText, System.Drawing.SystemColors.Window, new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0))));
            System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(NewFileDialog));
            this.listView1 = new System.Windows.Forms.ListView();
            this.dockingImageList = new System.Windows.Forms.ImageList(this.components);
            this.label1 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.textBox1 = new System.Windows.Forms.TextBox();
            this.okButton = new System.Windows.Forms.Button();
            this.cancelButton = new System.Windows.Forms.Button();
            this.SuspendLayout();
            // 
            // listView1
            // 
            this.listView1.Items.AddRange(new System.Windows.Forms.ListViewItem[] {
            listViewItem1,
            listViewItem2});
            this.listView1.LargeImageList = this.dockingImageList;
            this.listView1.Location = new System.Drawing.Point(8, 48);
            this.listView1.Name = "listView1";
            this.listView1.Size = new System.Drawing.Size(480, 160);
            this.listView1.SmallImageList = this.dockingImageList;
            this.listView1.TabIndex = 0;
            this.listView1.UseCompatibleStateImageBehavior = false;
            this.listView1.SelectedIndexChanged += new System.EventHandler(this.listView1_SelectedIndexChanged);
            // 
            // dockingImageList
            // 
            this.dockingImageList.ImageStream = ((System.Windows.Forms.ImageListStreamer)(resources.GetObject("dockingImageList.ImageStream")));
            this.dockingImageList.TransparentColor = System.Drawing.Color.Transparent;
            this.dockingImageList.Images.SetKeyName(0, "");
            this.dockingImageList.Images.SetKeyName(1, "");
            this.dockingImageList.Images.SetKeyName(2, "");
            this.dockingImageList.Images.SetKeyName(3, "");
            this.dockingImageList.Images.SetKeyName(4, "");
            this.dockingImageList.Images.SetKeyName(5, "");
            this.dockingImageList.Images.SetKeyName(6, "");
            this.dockingImageList.Images.SetKeyName(7, "");
            this.dockingImageList.Images.SetKeyName(8, "");
            this.dockingImageList.Images.SetKeyName(9, "");
            this.dockingImageList.Images.SetKeyName(10, "");
            this.dockingImageList.Images.SetKeyName(11, "");
            this.dockingImageList.Images.SetKeyName(12, "");
            this.dockingImageList.Images.SetKeyName(13, "");
            this.dockingImageList.Images.SetKeyName(14, "");
            this.dockingImageList.Images.SetKeyName(15, "");
            this.dockingImageList.Images.SetKeyName(16, "");
            this.dockingImageList.Images.SetKeyName(17, "");
            this.dockingImageList.Images.SetKeyName(18, "");
            this.dockingImageList.Images.SetKeyName(19, "");
            this.dockingImageList.Images.SetKeyName(20, "");
            this.dockingImageList.Images.SetKeyName(21, "");
            this.dockingImageList.Images.SetKeyName(22, "");
            this.dockingImageList.Images.SetKeyName(23, "");
            this.dockingImageList.Images.SetKeyName(24, "");
            this.dockingImageList.Images.SetKeyName(25, "");
            this.dockingImageList.Images.SetKeyName(26, "");
            this.dockingImageList.Images.SetKeyName(27, "");
            this.dockingImageList.Images.SetKeyName(28, "");
            this.dockingImageList.Images.SetKeyName(29, "");
            this.dockingImageList.Images.SetKeyName(30, "");
            this.dockingImageList.Images.SetKeyName(31, "");
            this.dockingImageList.Images.SetKeyName(32, "");
            this.dockingImageList.Images.SetKeyName(33, "");
            this.dockingImageList.Images.SetKeyName(34, "");
            this.dockingImageList.Images.SetKeyName(35, "");
            this.dockingImageList.Images.SetKeyName(36, "");
            this.dockingImageList.Images.SetKeyName(37, "");
            this.dockingImageList.Images.SetKeyName(38, "");
            this.dockingImageList.Images.SetKeyName(39, "");
            this.dockingImageList.Images.SetKeyName(40, "");
            this.dockingImageList.Images.SetKeyName(41, "");
            this.dockingImageList.Images.SetKeyName(42, "");
            this.dockingImageList.Images.SetKeyName(43, "");
            this.dockingImageList.Images.SetKeyName(44, "");
            this.dockingImageList.Images.SetKeyName(45, "");
            this.dockingImageList.Images.SetKeyName(46, "");
            this.dockingImageList.Images.SetKeyName(47, "");
            this.dockingImageList.Images.SetKeyName(48, "");
            this.dockingImageList.Images.SetKeyName(49, "");
            this.dockingImageList.Images.SetKeyName(50, "");
            this.dockingImageList.Images.SetKeyName(51, "");
            this.dockingImageList.Images.SetKeyName(52, "");
            this.dockingImageList.Images.SetKeyName(53, "");
            this.dockingImageList.Images.SetKeyName(54, "");
            this.dockingImageList.Images.SetKeyName(55, "");
            this.dockingImageList.Images.SetKeyName(56, "");
            this.dockingImageList.Images.SetKeyName(57, "");
            this.dockingImageList.Images.SetKeyName(58, "");
            this.dockingImageList.Images.SetKeyName(59, "");
            this.dockingImageList.Images.SetKeyName(60, "");
            this.dockingImageList.Images.SetKeyName(61, "");
            this.dockingImageList.Images.SetKeyName(62, "");
            this.dockingImageList.Images.SetKeyName(63, "");
            this.dockingImageList.Images.SetKeyName(64, "");
            this.dockingImageList.Images.SetKeyName(65, "");
            this.dockingImageList.Images.SetKeyName(66, "");
            this.dockingImageList.Images.SetKeyName(67, "");
            this.dockingImageList.Images.SetKeyName(68, "");
            this.dockingImageList.Images.SetKeyName(69, "");
            this.dockingImageList.Images.SetKeyName(70, "");
            this.dockingImageList.Images.SetKeyName(71, "");
            this.dockingImageList.Images.SetKeyName(72, "");
            this.dockingImageList.Images.SetKeyName(73, "");
            this.dockingImageList.Images.SetKeyName(74, "");
            this.dockingImageList.Images.SetKeyName(75, "");
            this.dockingImageList.Images.SetKeyName(76, "");
            this.dockingImageList.Images.SetKeyName(77, "");
            this.dockingImageList.Images.SetKeyName(78, "");
            this.dockingImageList.Images.SetKeyName(79, "");
            this.dockingImageList.Images.SetKeyName(80, "");
            // 
            // label1
            // 
            this.label1.Location = new System.Drawing.Point(8, 16);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(100, 23);
            this.label1.TabIndex = 1;
            this.label1.Text = "Templates";
            // 
            // label2
            // 
            this.label2.Location = new System.Drawing.Point(8, 224);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(64, 23);
            this.label2.TabIndex = 2;
            this.label2.Text = "FileName";
            // 
            // textBox1
            // 
            this.textBox1.Location = new System.Drawing.Point(8, 248);
            this.textBox1.Name = "textBox1";
            this.textBox1.Size = new System.Drawing.Size(472, 23);
            this.textBox1.TabIndex = 3;
            // 
            // okButton
            // 
            this.okButton.BackColor = System.Drawing.SystemColors.Control;
            this.okButton.Location = new System.Drawing.Point(312, 280);
            this.okButton.Name = "okButton";
            this.okButton.Size = new System.Drawing.Size(75, 23);
            this.okButton.TabIndex = 4;
            this.okButton.Text = "Ok";
            this.okButton.UseVisualStyleBackColor = false;
            this.okButton.Click += new System.EventHandler(this.okButton_Click);
            // 
            // cancelButton
            // 
            this.cancelButton.Location = new System.Drawing.Point(408, 280);
            this.cancelButton.Name = "cancelButton";
            this.cancelButton.Size = new System.Drawing.Size(75, 23);
            this.cancelButton.TabIndex = 5;
            this.cancelButton.Text = "Cancel";
            this.cancelButton.Click += new System.EventHandler(this.cancelButton_Click);
            // 
            // NewFileDialog
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(6, 16);
            this.ClientSize = new System.Drawing.Size(496, 334);
            this.Controls.Add(this.cancelButton);
            this.Controls.Add(this.okButton);
            this.Controls.Add(this.textBox1);
            this.Controls.Add(this.label2);
            this.Controls.Add(this.label1);
            this.Controls.Add(this.listView1);
            this.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F);
            this.Name = "NewFileDialog";
            this.Text = "NewFileDialog";
            this.ResumeLayout(false);
            this.PerformLayout();

        }
        #endregion

        private void listView1_SelectedIndexChanged(object sender, System.EventArgs e)
        {
            if (this.listView1.SelectedItems.Count > 0)
            {
                if (this.listView1.SelectedItems[0].Text != "")
                {
                    if (this.listView1.SelectedItems[0].Text == ".cs")
                    {
                        this.textBox1.Text = "Class1.cs";
                        this.textBox1.Focus();
                    }
                    else if (this.listView1.SelectedItems[0].Text == ".vb")
                    {
                        this.textBox1.Text = "Class1.vb";
                        this.textBox1.Focus();
                    }
                }
            }
        }

        private void okButton_Click(object sender, System.EventArgs e)
        {
            if (textBox1.Text != "")
            {
                if (textBox1.Text.EndsWith(".cs") || textBox1.Text.EndsWith(".vb"))
                {
                    if (!System.IO.File.Exists(System.IO.Path.GetDirectoryName(Application.ExecutablePath) + "\\" + textBox1.Text))
                    {
                        createNewFile();
                    }
                    else
                    {
                        DialogResult dr = MessageBox.Show("File already exists,Overwrite it??", "", MessageBoxButtons.OKCancel, MessageBoxIcon.Information);
                        if (dr == DialogResult.OK)
                        {
                            createNewFile();
                        }
                    }
                }
                else
                {
                    MessageBox.Show("Enter proper extension for File");
                }
            }
        }

        private void createNewFile()
        {
            try
            {
                Form activeMDIChild = this.mainForm.CheckFormInChildWindow(this.textBox1.Text);
                if (activeMDIChild != null)
                    activeMDIChild.Close();
                System.IO.File.Create(System.IO.Path.GetDirectoryName(Application.ExecutablePath) + "\\" + textBox1.Text);
                mainForm.openFileInCodeEditor(System.IO.Path.GetDirectoryName(Application.ExecutablePath) + "\\" + textBox1.Text);
                this.Close();
            }
            catch (Exception e)
            {
                MessageBox.Show(e.Message);
            }
        }

        private void cancelButton_Click(object sender, System.EventArgs e)
        {
            this.Close();
        }

        private void button1_Click(object sender, System.EventArgs e)
        {
            this.listView1.View = View.LargeIcon;
        }

        private void button2_Click(object sender, System.EventArgs e)
        {
            this.listView1.View = View.SmallIcon;
        }
    }
}
